airgun/entities/virtwho_configure.py

```diff
@@ -0,0 +1,121 @@
+from navmazing import NavigateToSibling
+
+from airgun.entities.base import BaseEntity
+from airgun.navigation import NavigateStep, navigator
+from airgun.views.virtwho_configure import (
+    VirtwhoConfigureCreateView,
+    VirtwhoConfigureEditView,
+    VirtwhoConfiguresView,
+    VirtwhoConfigureDetailsView,
+)
+
+
+class VirtwhoConfigureEntity(BaseEntity):
+
+    def _reset_values(self, values):
+        mapping = {
+            'esx': 'VMware vSphere / vCenter (esx)',
+            'xen': 'XenServer (xen)',
+            'hyperv': 'Microsoft Hyper-V (hyperv)',
+            'rhevm': 'Red Hat Virtualization Hypervisor (rhevm)',
+            'libvirt': 'libvirt',
+            'kubevirt': 'Container-native virtualization',
+        }
+        vals = values.copy()
+        if 'hypervisor_type' in vals:
+            hypervisor_value = vals['hypervisor_type']
+            vals.update({'hypervisor_type': mapping[hypervisor_value]})
+        return vals
+
+    def create(self, values):
+        """Create new virtwho configure entity"""
+        view = self.navigate_to(self, 'New')
+        values = self._reset_values(values)
+        view.fill(values)
+        view.submit.click()
+        view.flash.assert_no_error()
+        view.flash.dismiss()
+
+    def search(self, value):
+        """Search for virtwho configure"""
+        view = self.navigate_to(self, 'All')
+        return view.search(value)
+
+    def edit(self, name, values):
+        """Edit specific virtwho configure values"""
+        view = self.navigate_to(self, 'Edit', entity_name=name)
+        values = self._reset_values(values)
+        view.fill(values)
+        view.submit.click()
+        view.flash.assert_no_error()
+        view.flash.dismiss()
+
+    def read(self, entity_name, widget_names=None):
+        """Read all values for existing virtwho configure entity"""
+        view = self.navigate_to(self, 'Details', entity_name=entity_name)
+        return view.read(widget_names=widget_names)
+
+    def delete(self, value):
+        """Delete specific virtwho configure"""
+        view = self.navigate_to(self, 'All')
+        view.search(value)
+        view.table.row(name=value)['Actions'].widget.fill('Delete')
+        self.browser.handle_alert()
+        view.flash.assert_no_error()
+        view.flash.dismiss()
+
+
+@navigator.register(VirtwhoConfigureEntity, 'All')
+class ShowAllVirtwhoConfigures(NavigateStep):
+    """Navigate to All Activation Keys page"""
+    VIEW = VirtwhoConfiguresView
+
+    def step(self, *args, **kwargs):
+        self.view.menu.select('Infrastructure', 'Virt-who configurations')
+
```

Comment:
Are we hardcoding this here?

Reply:
Erm... why not? This is common idiom used in virtually every entity.

Reply:
I'v seen almost all the menu's element are located by this method, we don't need to use other xpath to locate it